### PR TITLE
[Boom] Add data type to BoomError #16956

### LINKED ISSUE #16956: boom: BoomError lacks data member
- [x] I tried using the `@types/boom` package and had problems.
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @AJamesPhillips 

You need `data?: any` in `BoomError`, else:

```
error TS2339: Property 'data' does not exist on type 'BoomError'.
```

To show `data` in action:

```
$ node
> require('boom').create(400, 'fnord', { a: 1 }).data
{ a: 1 }
```

### PR DESCRIPTION
**Pull request by AJamesPhillips**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: ** **closes #16956** **
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by AJamesPhillips** [comment]
@andy-ms any insight here would be appreciated.  These errors make no sense to be at the moment:


    === ERRORS ===
    Error in hapi
    Command failed: node /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/tester/../../node_modules/dtslint/bin/index.js --noLint
    ../boom/index.d.ts(15,37): error TS1005: ',' expected.
    ../boom/index.d.ts(15,39): error TS1139: Type parameter declaration expected.
    ../boom/index.d.ts(15,45): error TS1109: Expression expected.
    ../boom/index.d.ts(15,59): error TS1005: ';' expected.
    ../boom/index.d.ts(27,19): error TS1109: Expression expected.
    Package compiles in TypeScript 2.3 but not in 2.2.
    You can add a line '// TypeScript Version: 2.3' to the end of the header to specify a newer compiler version.
    ../boom/index.d.ts(15,37): error TS1005: ',' expected.
    ../boom/index.d.ts(15,39): error TS1139: Type parameter declaration expected.
    ../boom/index.d.ts(15,45): error TS1109: Expression expected.
    ../boom/index.d.ts(15,59): error TS1005: ';' expected.
    ../boom/index.d.ts(27,19): error TS1109: Expression expected.
    Package compiles in TypeScript 2.3 but not in 2.2.
    You can add a line '// TypeScript Version: 2.3' to the end of the header to specify a newer compiler version.
    The following packages had errors: hapi

It's clearly saying hapi has errored but then giving and error in the boom/index.d.ts with error values that don't make sense as it compiles on my local machine with typescript 2.3 and also then says to add `// TypeScript Version: 2.3` to header when it's already there and it works with 2.3 ...? 😕

**Comment by andy-ms** [comment]
`types/hapi/index.d.ts` still has `// TypeScript Version: 2.2`.

**Comment by andy-ms** [comment]
Or just rebase now that #16802 is in.

**Comment by AJamesPhillips** [comment]
* forehead slap * thanks @andy-ms .  This is ready to merge.

File: types/boom/v3/index.d.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Project: http://github.com/hapijs/boom
 // Definitions by: Igor Rogatty <http://github.com/rogatty>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
```
Comment (andy-ms) [inline types/boom/v3/index.d.ts]: This shouldn't be necessary if you didn't change the v3 types.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**